ToopherDotNet/ToopherDotNet.cs

```diff
@@ -8,334 +9,841 @@
 		}
 
-		// Check on status of a pairing request
-		// 
-		// Must provide the ID returned when the pairing request was initiated
-		public PairingStatus GetPairingStatus (string pairingRequestId)
+		/// <summary>
+		/// Generate a URL to retrieve a Toopher Pairing iFrame for a given user.
+		/// </summary>
+		/// <param name="userName">Unique name that identifies this user. This will be displayed to the user on their mobile device when they pair or authenticate.</param>
+		/// <param name="resetEmail">Email address that the user has access to. In case the user has lost or cannot access their mobile device, Toopher will send a reset email to this address.</param>
+		/// <param name="extras">An optional Dictionary of extra parameters to provide to the API.</param>
+		/// <returns>A string URL that can be used to retrieve the Pairing iFrame by the user's browser.</reeturns>
+		public string GetUserManagementUrl(string userName, string resetEmail, Dictionary<string, string> extras = null)
 		{
-			string endpoint = String.Format ("pairings/{0}", pairingRequestId);
+			NameValueCollection parameters = new NameValueCollection();
+			long ttl;
+			if (extras != null && extras.ContainsKey("ttl"))
+			{
+				ttl = Convert.ToInt64(extras["ttl"]);
+				extras.Remove("ttl");
+			} else
+			{
+				ttl = DEFAULT_TTL;
+			}
+
+			parameters.Add("v", IFRAME_VERSION);
+			parameters.Add("username", userName);
+			parameters.Add("reset_email", resetEmail);
+			parameters.Add("expires", (GetUnixEpochTimeInSeconds() + ttl).ToString());
 
-			var json = get (endpoint);
-			return new PairingStatus (json);
+			if (extras != null)
+			{
+				foreach (KeyValuePair<string, string> kvp in extras)
+				{
+					parameters.Add(kvp.Key, kvp.Value);
+				}
+			}
+			return GetOauthUrl(baseUrl + "web/manage_user", parameters);
 		}
 
-		// Authenticate an action with Toopher
-		//
-		// Provide pairing ID, a name for the terminal (displayed to user) and
-		// an option action name (displayed to user) [defaults to "log in"]
-		public AuthenticationStatus Authenticate (string pairingId, string terminalName, string actionName = null, Dictionary<string, string> extras = null)
+		/// <summary>
+		/// Verify the authenticity of data returned from the Toopher iFrame by validating the crytographic signature.
+		/// </summary>
+		/// <param name="params">The data returned from the iFrame.</param>
+		/// <param name="sessionToken">The session token</param>
+		/// <param name="ttl">Time-To-Live (seconds) to enforce on the Toopher API signature. This value sets the maximum duration between the Toopher API creating the signature and the signature being validated on your server.</param>
+		/// <returns>A Dictionary of the validated data if the signature is valid, or null if the signature is invalid.</returns>
+		public Dictionary<string, string> ValidatePostback(Dictionary<string, string[]> parameters, string sessionToken, long ttl)
```

Comment:
Also, can you add an overload that takes `parameters` as a `Dictionary<string, string>`?